node/middlewares/validate.ts

```
@@ -0,0 +1,24 @@
+import { UserInputError } from '@vtex/api'
+
+export async function validate(ctx: Context, next: () => Promise<any>) {
+  const {
+    vtex: {
+      route: { params },
+    },
+  } = ctx
+
+  console.log('Received params:', params)
+
+  const { id } = params
+
+  if (!id) {
```

I think you don't need this. Because this middleware will only be called in case it receives an id as a param. You can throw this error in the parsing code.